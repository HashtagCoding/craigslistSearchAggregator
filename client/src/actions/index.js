import axios from 'axios'

export const getCraigslistFeed = (dispatch) => {
  return dispatch => axios.get('/craigslist_scraper')
    .then(res => {
      console.log('res2 =>', res) 
      return res
    })
    .then(
      data => dispatch({ type: 'GET_CRAIGSLIST_FEED', data}),
      err => dispatch({ type: 'GET_CRAIGSLIST_FEED_ERROR', err })
    )
}

export const getTweeterFeed = () => {
  return {
    type: 'GET_TWITTER_FEED'
  }
}
