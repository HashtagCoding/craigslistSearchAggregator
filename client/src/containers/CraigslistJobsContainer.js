import { connect } from 'react-redux';
import { getCraigslistJobs } from '../actions';
import CraigslistJobsList from '../components/CraigslistJobsList';

const mapStateToProps = (state) => {
  return {
    list: state.craigslistJobs,
  };
};

const mapDispatchToProps = (dispatch) => {
  return {
    onClick_getCraigslistJobs: (jobType, jobCategory) => {
      dispatch(getCraigslistJobs(jobType, jobCategory));
    },
  };
};

const CraigslistJobsContainer = connect(
  mapStateToProps,
  mapDispatchToProps,
)(CraigslistJobsList);

export default CraigslistJobsContainer;
