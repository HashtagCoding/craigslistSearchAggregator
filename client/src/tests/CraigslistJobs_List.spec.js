import React from 'react'
import { shallow, mount } from 'enzyme';
import renderer from 'react-test-renderer'
import configureStore from 'redux-mock-store'
import {Provider} from 'react-redux'
import {createStore} from 'redux'
import toJson from 'enzyme-to-json'

import CraigslistJobsContainer from '../containers/CraigslistJobsContainer'
import CraigslistJobs_Item from '../components/CraigslistJobs_List'
import { getCraigslistJobs } from '../actions'

import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-15';
configure({ adapter: new Adapter() });


describe('CraigslistJobs_Item rendering correctly', () => {
  const component = shallow(<CraigslistJobs_Item />)
  const tree = toJson(component)

  it('snapshot matches', () => {
    expect(tree).toMatchSnapshot()
    // expect(tree2).toMatchSnapshot()
  })

});
