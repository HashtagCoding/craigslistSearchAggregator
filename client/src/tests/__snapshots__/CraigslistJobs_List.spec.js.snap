// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CraigslistJobs_Item should render as expected snapshot matches 1`] = `
<Grid
  centered={true}
  divided="vertically"
>
  <GridRow
    columns={1}
    stretched={true}
    textAlign="left"
  />
</Grid>
`;
